### ISSUE TITLE
single file app .r vs .R

### ISSUE THREAD

**Issue by fxi** [rstudio/shiny-server#112]
Shiny Server did not found my app.R application.
Everything works as expected with a file named app.r
(shiny server v1.3.0.661, Node.js v0.10.21,Ubuntu 14.04.2 LTS)
Maybe something linked with: 
https://github.com/rstudio/shiny-server/search?utf8=%E2%9C%93&q=singleApp&type=Code
Thanks,
F

**Comment by aronatkins** [comment]
I have not been able to duplicate this problem. I configured both multi-application and single-application locations with both `app.R` and `app.r` alternatives; all applications properly started.

I am testing with shiny server 1.3.0.661 and shiny 0.11.1
```
# cat /opt/shiny-server/VERSION
1.3.0.661
```

```
# R -q -e 'packageVersion("shiny")'
> packageVersion("shiny")
[1] ‘0.11.1’
>
>
```

The relevant bits from `shiny-server.conf`:
```
  location /upper {
    app_dir /opt/shiny-server/upper;
    log_dir /var/log/shiny-server;
  }
  location /lower {
    app_dir /opt/shiny-server/lower;
    log_dir /var/log/shiny-server;
  }
  location /cases {
    site_dir /opt/shiny-server/cases;
    log_dir /var/log/shiny-server;
  }
```

The directory structure of these locations:
```
cases
├── lower
│   └── app.r
└── upper
    └── app.R
upper
└── app.R
lower
└── app.r
```
In my testing, all files and directories were owned by `root`. Directories had `0755` mode and files had `0644` mode.

The contents of `app.r` / `app.R`:
```
library(shiny)

server <- function(input, output) {}

ui <- fluidPage("hello, shiny")

shinyApp(ui = ui, server = server)
```

Could you help us understand what else might be different about your configuration? What version of Shiny is being used? Could you tell me the ownership and permissions of both your `app.r` and `app.R` files? The `location` section from your `shiny-server.conf` file might also help narrow my testing.

**Comment by wch** [comment]
@fxi Do you have any other files in the directory?

**Comment by fxi** [comment]
Ok.

This happen after the transition from a two files app to a single file app, but this was not linked to extension case, sorry for the false issue! 

I was able to reproduce the issue by checking out a revision where I was using a server.r/ui.r app, then checking a revision with app.r and reloaded my application.

So a sequence like this:

Loaded my shiny app with these files:
```{sh}
.
├── config.R
├── demo
├── loadlib.R
├── modules
├── packrat
├── README.md
├── server.R
├── tools
├── ui.R
└── www
```
Then checked out and reloaded a revision where only app.r was used  : 
```{sh}
.
├── app.R
├── config.R
├── demo
├── loadlib.R
├── modules
├── packrat
├── README.md
├── tools
└── www
```
Then then error was:
![screen shot 2015-04-30 at 12 18 05](https://cloud.githubusercontent.com/assets/1196833/7411131/5c163494-ef35-11e4-97ea-09e4ce47bcf5.png)

So, the old session was still active and looked for ui.R, but the message mentioned that shiny also did not found app.R when app.R was present.

After reloading a second time, the error was gone.

Thanks for your time and thank you very much for maintaining shiny, that's a wonderful tool !

**Comment by jcheng5** [comment]
I kind of vaguely recall that this happens when you delete an entire app
directory while a Shiny process is active. Its process working directory is
the old directory that you deleted. Would that explain it in your case?

**Comment by fxi** [comment]
Yes, this could explain what happened: in one revision, I used server/ui files, and in the next revision, I've changed to single file app. As I use git to update my app, with a page reload to switch to the new revision, I naively expected that shiny could figure out how to manage the new configuration. The only minor issue was the misleading message thrown by shiny: "Shiny couldn't find any UI for this application. We looked in: www/index.html, ui.R, app.R". In reality, shiny process was not able to find anything, as the server/ui component used in the process went to another file. So, this is a tiny problem.
